### PR TITLE
Fix event parentchanged

### PR DESCRIPTION
**Pull request by pl-buiquang**
Signed-off-by: Paul Bui-Quang <paul.buiquang@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
feature

**What is the current behavior?** *(You can also link to an open issue here)*
ParentChanged node event is missing the new parent id information

**What is the new behavior (if this is a feature change)?**
ParentChanged node event now have the new parent id

**Other information**:
Add a updateIncludes utility method on AbstractScript

### CODE REVIEW

**Comment by pl-buiquang** [comment]
> An open question : I wonder if we should bother with version compatibility issues or not.
> If the server and the client don't update at the same time, I fear the communication will not work any more (probably some exceptions when reading the JSON).

That's why I usually configure the objectmapper with
`configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);`
Maybe we should do that

File: afs-storage-api/src/main/java/com/powsybl/afs/storage/events/ParentChanged.java [hunk 1]
```diff
@@ -16,9 +16,17 @@
 
     public static final String TYPENAME = "PARENT_CHANGED";
 
+    @JsonProperty("newParentId")
+    private String newParentId;
+
     @JsonCreator
-    public ParentChanged(@JsonProperty("id") String id) {
+    public ParentChanged(@JsonProperty("id") String id, @JsonProperty("newParentId") String newParentId) {
```
Comment (mathbagu) [inline afs-storage-api/src/main/java/com/powsybl/afs/storage/events/ParentChanged.java:23]: From my point of view, you should (if possible) add both `oldParentId` and `newParentId`.

File: afs-storage-api/src/main/java/com/powsybl/afs/storage/events/ParentChanged.java [hunk 2]
```diff
@@ -16,9 +16,17 @@
...
         super(id, TYPENAME);
+        this.newParentId = newParentId;
```
Comment (mathbagu) [inline afs-storage-api/src/main/java/com/powsybl/afs/storage/events/ParentChanged.java:25]: Is it possible that this object has no new parent? In that case, you should assert that it's not null.

Comment (pl-buiquang) [reply afs-storage-api/src/main/java/com/powsybl/afs/storage/events/ParentChanged.java:25]: It is not possible for this event not to have a newParentId

File: afs-storage-api/src/main/java/com/powsybl/afs/storage/events/ParentChanged.java [hunk 3]
```diff
@@ -9,34 +9,63 @@
...
+    private final String newParentId;
+
     @JsonCreator
-    public ParentChanged(@JsonProperty("id") String id) {
+    public ParentChanged(@JsonProperty("id") String id, @JsonProperty("oldParentId") String oldParentId, @JsonProperty("newParentId") String newParentId) {
         super(id, TYPENAME);
+        Objects.requireNonNull(oldParentId);
+        Objects.requireNonNull(newParentId);
+        this.oldParentId = oldParentId;
+        this.newParentId = newParentId;
```
Comment (mathbagu) [inline afs-storage-api/src/main/java/com/powsybl/afs/storage/events/ParentChanged.java:33]: This kind of lines should be written like this, to be more compact:
```
this.oldParentId = Objects.requireNonNull(oldParentId);
this.newParentId = Objects.requireNonNull(newParentId);
```

Comment (pl-buiquang) [reply afs-storage-api/src/main/java/com/powsybl/afs/storage/events/ParentChanged.java]: indeed

**Pull request merged**